compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNonTerminator.java

```diff
@@ -872,6 +872,8 @@ public static class FPLoad extends BIRNonTerminator {
         public List<BIROperand> closureMaps;
         public BType type;
         public boolean isWorker;
+        public BType enclosedType = null;
+        public String fieldName;
```

Comment:
This is added because we need the type information at the `FPLoad` of default value functions.
Because, at the code generation of the default value FPLoad, we pass the closure information to the record type. This is done to support the closure with inline records, in which case the closure name is not enough. https://github.com/ballerina-platform/ballerina-lang/issues/40214

Reply:
Can't it be derived from the name? Atm, it looks like we are kinda polluting the FPLoad instruction. FPLoad is just an FPLoad and it doesn't have to be know whether it is one corresponding to a default value.

Reply:
We need the type and the field name when we create the Function Pointer at runtime. Because the closure values are dynamically handled using Java `Function` interfaces, created with the new Function Pointers. So, we can't get the  Function Pointers instance from the type.

I agree that having the information inside FPLoad is not preferred. We may need to add a new BIR instruction to handle this case. We earlier implemented that approach and removed it during the review as the code generation is very similar to the `FPLoad` instruction.
https://github.com/ballerina-platform/ballerina-lang/pull/40579/commits/e15f521e3cd2b29ce6c07be815af32ade42ac1e2